include/deal.II/non_matching/mapping_info.h

```diff
@@ -157,30 +293,550 @@ namespace NonMatching
+
+            const unsigned int current_face_index =
+              cell_index_offset[cell_index] + f;
+
+            auto it =
+              unit_points.begin() + unit_points_index[current_face_index];
+            for (const auto &unit_point : unit_points_on_cell)
+              {
+                *it = unit_point;
+                ++it;
+              }
+
+            compute_mapping_data_for_face_quadrature(
+              cell, f, quadrature_on_face, mapping_data[current_face_index]);
+          }
+        if (do_cell_index_compression)
+          cell_index_to_compressed_cell_index[cell->active_cell_index()] =
+            cell_index;
+
+        ++cell_index;
+      }
+
+    state = State::faces_on_cells_in_vector;
   }
 
 
 
   template <int dim, int spacedim>
-  const std::vector<Point<dim>> &
-  MappingInfo<dim, spacedim>::get_unit_points() const
+  const std::vector<Point<dim>>
+  MappingInfo<dim, spacedim>::get_unit_points(
+    const unsigned int cell_index,
+    const unsigned int face_number) const
   {
-    return unit_points;
+    if (cell_index == numbers::invalid_unsigned_int &&
+        face_number == numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::single_cell,
+               ExcMessage(
+                 "This mapping info is not reinitialized for a single cell"));
+        return unit_points;
+      }
+    else if (face_number == numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::cell_vector,
+               ExcMessage(
+                 "This mapping info is not reinitialized for a cell vector"));
+        if (do_cell_index_compression)
+          {
+            Assert(
+              cell_index_to_compressed_cell_index[cell_index] !=
+                numbers::invalid_unsigned_int,
+              ExcMessage(
+                "Mapping info object was not initialized for this active cell "
+                "index"));
+            const auto it_begin =
+              unit_points.begin() +
+              unit_points_index
+                [cell_index_to_compressed_cell_index[cell_index]];
+            const auto it_end =
+              unit_points.begin() +
+              unit_points_index
+                [cell_index_to_compressed_cell_index[cell_index] + 1];
+            return std::vector<Point<dim>>(it_begin, it_end);
+          }
+        else
+          {
+            const auto it_begin =
+              unit_points.begin() + unit_points_index[cell_index];
+            const auto it_end =
+              unit_points.begin() + unit_points_index[cell_index + 1];
+            return std::vector<Point<dim>>(it_begin, it_end);
+          }
+      }
+    else if (cell_index != numbers::invalid_unsigned_int)
+      {
+        Assert(
+          state == State::faces_on_cells_in_vector,
+          ExcMessage(
+            "This mapping info is not reinitialized for faces on cells in a "
+            "vector"));
```

Comment:
I don't like when `clang-format` creates so many bad line breaks. Can you try what happens if you break the line earlier, say before `faces`?

Reply:
Much better now 👍🏽